--- a/ziskos/entrypoint/src/lib.rs
+++ b/ziskos/entrypoint/src/lib.rs
@@ -110,6 +110,7 @@ mod ziskos {
     use crate::ziskos_definitions::ziskos_config::*;
     use core::arch::asm;
 
+    #[cfg(not(feature = "no_entrypoint"))]
     #[no_mangle]
     #[link_section = ".text.init"]
     unsafe extern "C" fn _start() -> ! {
@@ -163,6 +164,7 @@ mod ziskos {
         getrandom::register_custom_getrandom!(zkvm_getrandom);
     }
 
+    #[cfg(not(feature = "no_entrypoint"))]
     #[no_mangle]
     unsafe extern "C" fn _zisk_main() {
         {
@@ -236,6 +238,12 @@ mod ziskos {
     extern "C" fn sys_argv() {
         unimplemented!("sys_argv");
     }
+
+    #[no_mangle]
+    extern "C" fn sys_panic(_msg_ptr: *const u8, _msg_len: usize) -> ! {
+        // Empty implementation - panic handling is done by bflat runtime
+        loop {}
+    }
 
     #[no_mangle]
     pub unsafe extern "C" fn sys_alloc_aligned(bytes: usize, align: usize) -> *mut u8 {
@@ -282,9 +290,12 @@ mod ziskos {

         ptr
     }
+    #[cfg(not(feature = "no_entrypoint"))]
     core::arch::global_asm!(include_str!("dma/memcpy.s"));
+    #[cfg(not(feature = "no_entrypoint"))]
     core::arch::global_asm!(include_str!("dma/memmove.s"));
+    #[cfg(not(feature = "no_entrypoint"))]
     core::arch::global_asm!(include_str!("dma/memcmp.s"));
-    //core::arch::global_asm!(include_str!("dma/inputcpy.s"));
+    #[cfg(not(feature = "no_entrypoint"))]
     core::arch::global_asm!(include_str!("dma/memset.s"));
 }
